force-app/main/default/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,171 @@
+    */
+    public override void beforeDelete() {
+        // Prevent deletion of closed Opportunities
+        for (Opportunity oldOpp : this.oldOppList){
+            if (oldOpp.IsClosed){
+                oldOpp.addError('Cannot delete closed opportunity');
+            }
+        }
+    }
+
+ 
+    public override void afterInsert() {
+        // Create a new Task for newly inserted Opportunities
+        List<Task> oppTasks = new List<Task>();
+        for (Opportunity opp : this.newOppList){
+            Task tsk = new Task();
+            tsk.Subject = 'Call Primary Contact';
+            tsk.WhatId = opp.Id;
+            tsk.WhoId = opp.Primary_Contact__c;
+            tsk.OwnerId = opp.OwnerId;
+            tsk.ActivityDate = Date.today().addDays(3);
+            oppTasks.add(tsk);
+        }
+        insert oppTasks;
+    }
+
+
+    public override void afterUpdate() {
+        // Append Stage changes in Opportunity Description
+        List<Opportunity> updateOpps = new List<Opportunity>();
+        for(Opportunity opp : this.newOppList){
+            Opportunity oppOld = this.oldOppMap.get(opp.Id);
+            if (oppOld.StageName != null && oppOld.StageName != opp.StageName){
+                Opportunity newOpp = new Opportunity(Id = opp.Id);
+                newOpp.Description += '\n Stage Change:' + opp.StageName + ':' + DateTime.now().format();
+                updateOpps.add(newOpp);
+            }            
+        }
+        update updateOpps;
+    }
+
+
+    public override void afterDelete() {
+        notifyOwnersOpportunityDeleted(this.oldOppList);
+    }
+
+    /*
+    - Assigns a primary contact with the title of 'VP Sales' to undeleted Opportunities.
+    - Only updates the Opportunities that don't already have a primary contact.
+    */  
+    public override void afterUndelete() {    
+        Set<Id> oppAccountIds = new Set<Id>(); //create set of unduplciated Opportunity AccountIds to use to query related Contacts
+        for (Opportunity opp : this.newOppMap.values()){
+            if (opp.AccountId != null){
+                oppAccountIds.add(opp.AccountId);
+            }
+        }
+        List<Contact> accContacts = [SELECT Id, AccountId, Name, Title FROM Contact WHERE AccountId IN :oppAccountIds AND Title = 'VP Sales'];
+        Map<Id, Contact> accIdToContact = new Map<Id, Contact>(); //createa a map of AccountIds and a single related Contact that matches the critieria
+        for(Contact con : accContacts){
+            accIdToContact.put(con.AccountId, con);
+        }
+        List<Opportunity> oppsToUpdate = new List<Opportunity>(); // create a list of Opportunities to populate with updates
+        for(Opportunity opp : this.newOppMap.values()){
+            Id oppAccountId = opp.AccountId; // get the current Opportunity's AccountId to match to the AccountId/Contact map
+            if (opp.AccountId != null && accIdToContact.containsKey(opp.AccountId) && opp.Primary_Contact__c == null){
```

Comment:
You can either use the variable oppAccountId here or remove that variable altogether